drivers/console/rtt_console.c

```diff
@@ -18,14 +18,63 @@
 extern void __printk_hook_install(int (*fn)(int));
 extern void __stdout_hook_install(int (*fn)(int));
 
+static bool host_present;
+
+/** @brief Wait for fixed period.
+ *
+ */
+static void wait(void)
+{
+	if (k_is_in_isr()) {
+		k_busy_wait(1000*CONFIG_RTT_TX_RETRY_DELAY_MS);
+	} else {
+		k_sleep(CONFIG_RTT_TX_RETRY_DELAY_MS);
+	}
+}
+
 static int rtt_console_out(int character)
 {
 	unsigned int key;
 	char c = (char)character;
+	unsigned int cnt;
+	int watchdog_counter = CONFIG_RTT_TX_RETRY_CNT;
```

Comment:
this is a bit long as a variable name? maybe 'cnt_max'?

Reply:
changed.